cub/device/dispatch/dispatch_reduce.cuh

```diff
@@ -55,788 +54,1247 @@ CUB_NAMESPACE_BEGIN
+
+    // SegmentedReducePolicy
+    using SegmentedReducePolicy = ReducePolicy;
+  };
+
+  /// SM60
+  struct Policy600 : ChainedPolicy<600, Policy600, Policy350>
+  {
+    static constexpr int threads_per_block  = 256;
+    static constexpr int items_per_thread   = 16;
+    static constexpr int items_per_vec_load = 4;
+
+    // ReducePolicy (P100: 591 GB/s @ 64M 4B items; 583 GB/s @ 256M 1B items)
+    using ReducePolicy = AgentReducePolicy<threads_per_block,
+                                           items_per_thread,
+                                           AccumT,
+                                           items_per_vec_load,
+                                           BLOCK_REDUCE_WARP_REDUCTIONS,
+                                           LOAD_LDG>;
+
+    // SingleTilePolicy
+    using SingleTilePolicy = ReducePolicy;
+
+    // SegmentedReducePolicy
+    using SegmentedReducePolicy = ReducePolicy;
+  };
+
+  using MaxPolicy = Policy600;
 };
 
 
 
 /******************************************************************************
  * Single-problem dispatch
- ******************************************************************************/
+ *****************************************************************************/
 
 /**
- * Utility class for dispatching the appropriately-tuned kernels for device-wide reduction
+ * @brief Utility class for dispatching the appropriately-tuned kernels for 
+ *        device-wide reduction
+ *
+ * @tparam InputIteratorT
+ *   Random-access input iterator type for reading input items \iterator
+ *
+ * @tparam OutputIteratorT
+ *   Output iterator type for recording the reduced aggregate \iterator
+ *
+ * @tparam OffsetT
+ *   Signed integer type for global offsets
+ *
+ * @tparam ReductionOpT
+ *   Binary reduction functor type having member 
+ *   `auto operator()(const T &a, const U &b)`
+ *
+ * @tparam InitT 
+ *   Initial value type
  */
 template <
-    typename InputIteratorT,    ///< Random-access input iterator type for reading input items \iterator
-    typename OutputIteratorT,   ///< Output iterator type for recording the reduced aggregate \iterator
-    typename OffsetT,           ///< Signed integer type for global offsets
-    typename ReductionOpT,      ///< Binary reduction functor type having member <tt>T operator()(const T &a, const T &b)</tt>
-    typename OutputT =          ///< Data type of the output iterator
-        cub::detail::non_void_value_t<
-          OutputIteratorT,
-          cub::detail::value_t<InputIteratorT>>,
-    typename SelectedPolicy = DeviceReducePolicy<
-        cub::detail::value_t<InputIteratorT>,
-        OutputT,
-        OffsetT,
-        ReductionOpT> >
-struct DispatchReduce :
-    SelectedPolicy
+  typename InputIteratorT,
+  typename OutputIteratorT,
+  typename OffsetT,
+  typename ReductionOpT,
+  typename InitT = 
+    cub::detail::non_void_value_t<
+      OutputIteratorT, 
+      cub::detail::value_t<InputIteratorT>>,
+  typename AccumT = 
```

Comment:
Can you add the `release: breaking change` label and put a description of these changes to the Dispatch interface in the PR description? That way I'll make sure to call these changes out in the release notes.

Same goes for all of the accumulator-type changes in behavior -- I check that label when I'm building relnotes from the list of PRs.